read-fonts/src/tables/layout.rs

```diff
@@ -255,6 +285,44 @@ impl CoverageFormat2<'_> {
         }
     }
 
+    /// Returns the intersection of this table and input 'glyphs' set.
+    #[cfg(feature = "std")]
+    fn intersect_set(&self, glyphs: &IntSet<GlyphId>) -> IntSet<GlyphId> {
+        let range_count = self.range_count() as u32;
+        let num_bits = 32 - range_count.leading_zeros();
+        if range_count > (glyphs.len() as u32) * num_bits {
+            glyphs
+                .iter()
+                .filter_map(|g| self.get(g).map(|_| g))
+                .collect()
+        } else {
+            let mut out = IntSet::empty();
+            let mut last = GlyphId16::from(0);
+            for record in self.range_records() {
+                // break out of loop for overlapping/broken tables
+                let start_glyph = record.start_glyph_id();
+                if start_glyph < last {
+                    break;
+                }
+                let end = record.end_glyph_id();
+                last = end;
+
+                let start = GlyphId::from(start_glyph);
+                if glyphs.contains(start) {
+                    out.insert(start);
+                }
+
+                for g in glyphs.iter_after(start) {
```

Comment:
Nothing to do yet, but I'll add an iter_range() method to IntSet that iterates all values within the range (inclusive) that should simplify cases like this throughout this PR by not needing to check the start value separately.

Reply:
Yeah that will be very useful. Thanks!